src/backend/renderer.js

```diff
@@ -577,13 +574,21 @@ export function attach(
   }
 
   let pendingOperations: Array<number> = [];
-  let pendingOperationsQueue: Array<Array<number>> | null = [];
+  let pendingRealUnmountedIDs: Array<number> = [];
+  let pendingSimulatedUnmountedIDs: Array<number> = [];
+  let pendingOperationsQueue: Array<Uint32Array> | null = [];
 
+  // We keep track of which Fibers have been reported as unmounted by React
+  // during this commit phase so that we don't try to "hide" them or their
+  // children when Suspense flips to fallback. These Fibers won't have IDs.
+  let fibersUnmountedInThisCommitPhase: WeakSet<Fiber> = new WeakSet();
+
+  // TODO: we could make this layer DEV-only and write directly to pendingOperations.
```

Comment:
I don't think I understand this comment.

Reply:
Now that it’s append-only, the intermediate array doesn’t give us anything except extra validation. We could have written into the “operations” instead.